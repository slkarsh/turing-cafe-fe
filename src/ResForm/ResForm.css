input {
    height: 30px;
    width: 170px;
    margin: 15px;
}

button {
    height: 30px;
    width: 170px;
    background: teal;
    color: white
}