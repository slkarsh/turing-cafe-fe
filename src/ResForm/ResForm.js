import React, { Component } from 'react'

class ResForm extends Component {
    constructor() {
        super();
        this.state = {
            name: '',
            date: '',
            time: '',
            guests: 0,
        }
    }

    handleChange = e => {
        if ([e.target.name] === 'guests') {
            parseInt([e.target.value])
        }
        this.setState({ [e.target.name] : e.target.value})
    }

    createReservation = (event) => {
        event.preventDefault();
        const newResy = {
            id: Date.now(),
            name: this.state.name,
            date: this.state.date,
            time: this.state.time,
            number: parseInt(this.state.guests)
        }
        this.props.addReservation(newResy)
        this.resetInputs();
    }

    resetInputs = () => {
        this.setState({
            name: '',
            date: '',
            time: '',
            guests: 0
        })
    }


    render() {
        return (
            <form className='res-form'>
            <input
                className='name-input'
                type='text'
                placeholder='Name'
                value={this.state.name}
                name='name'
                onChange={(e) => this.handleChange(e)}
            />
            <input
                className='date-input'
                type='text'
                placeholder='Date (MM/DD)'
                value={this.state.date}
                name='date'
                onChange={(e) => this.handleChange(e)}
             />
             <input
                className='time-input'
                type='text'
                placeholder='Time'
                value={this.state.time}
                name='time'
                onChange={(e) => this.handleChange(e)}
             />
             <input
                className='guests-input'
                type='number'
                placeholder='Number of guests'
                value={this.state.guests}
                name='guests'
                onChange={(e) => this.handleChange(e)}
             />
             <button onClick={(event) => this.createReservation(event)}>Make Reservation</button>
            </form>
        )   
    }
}

export default ResForm;