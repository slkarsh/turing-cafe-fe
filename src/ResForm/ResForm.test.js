import React from 'react'
import { shallow } from 'enzyme'
import ResForm from './ResForm'


describe('ResForm', () => {
let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ResForm addReservation={jest.fn()}/>)
    })

it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
})

it('should update state when handleChange is called', () => {
    const mockEvent = {target: {name: 'name', value: 'Ollie'}}
    const expected = 'Ollie'

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('name')).toEqual(expected)
})

it('should reset state when resetInputs is called', () => {
    const expected
})
})