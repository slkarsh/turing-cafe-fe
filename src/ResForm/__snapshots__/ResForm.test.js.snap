// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResForm should match the snapshot 1`] = `
<form
  className="res-form"
>
  <input
    className="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="date-input"
    name="date"
    onChange={[Function]}
    placeholder="Date (MM/DD)"
    type="text"
    value=""
  />
  <input
    className="time-input"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="guests-input"
    name="guests"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value={0}
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
